src/common/marlin_vars.c

```diff
@@ -177,74 +188,102 @@ void marlin_vars_set_var(marlin_vars_t *vars, uint8_t var_id, variant8_t var) {
             break;
         case MARLIN_VAR_TEMP_BED:
-            sprintf(str, "%.1f", (double)(vars->temp_bed));
+            snprintf(str, size, "%.1f", (double)(vars->temp_bed));
             break;
         case MARLIN_VAR_TTEM_NOZ:
-            sprintf(str, "%.1f", (double)(vars->target_nozzle));
+            snprintf(str, size, "%.1f", (double)(vars->target_nozzle));
             break;
         case MARLIN_VAR_TTEM_BED:
-            sprintf(str, "%.1f", (double)(vars->target_bed));
+            snprintf(str, size, "%.1f", (double)(vars->target_bed));
             break;
         case MARLIN_VAR_Z_OFFSET:
-            sprintf(str, "%.4f", (double)(vars->z_offset));
+            snprintf(str, size, "%.4f", (double)(vars->z_offset));
             break;
         case MARLIN_VAR_FANSPEED:
-            sprintf(str, "%u", (unsigned int)(vars->fan_speed));
+            snprintf(str, size, "%u", (unsigned int)(vars->fan_speed));
             break;
         case MARLIN_VAR_PRNSPEED:
-            sprintf(str, "%u", (unsigned int)(vars->print_speed));
+            snprintf(str, size, "%u", (unsigned int)(vars->print_speed));
             break;
         case MARLIN_VAR_FLOWFACT:
-            sprintf(str, "%u", (unsigned int)(vars->flow_factor));
+            snprintf(str, size, "%u", (unsigned int)(vars->flow_factor));
             break;
         case MARLIN_VAR_WAITHEAT:
-            sprintf(str, "%u", (unsigned int)(vars->wait_heat));
+            snprintf(str, size, "%u", (unsigned int)(vars->wait_heat));
             break;
         case MARLIN_VAR_WAITUSER:
-            sprintf(str, "%u", (unsigned int)(vars->wait_user));
+            snprintf(str, size, "%u", (unsigned int)(vars->wait_user));
             break;
         case MARLIN_VAR_SD_PRINT:
-            sprintf(str, "%u", (unsigned int)(vars->sd_printing));
+            snprintf(str, size, "%u", (unsigned int)(vars->sd_printing));
             break;
         case MARLIN_VAR_SD_PDONE:
-            sprintf(str, "%u", (unsigned int)(vars->sd_percent_done));
+            snprintf(str, size, "%u", (unsigned int)(vars->sd_percent_done));
             break;
         case MARLIN_VAR_DURATION:
-            sprintf(str, "%lu", (long unsigned int)(vars->print_duration));
+            snprintf(str, size, "%lu", (long unsigned int)(vars->print_duration));
+            break;
+        case MARLIN_VAR_MEDIAINS:
+            snprintf(str, size, "%u", (unsigned int)(vars->media_inserted));
+            break;
+        case MARLIN_VAR_PRNSTATE:
+            snprintf(str, size, "%u", (unsigned int)(vars->print_state));
+            break;
+        case MARLIN_VAR_FILENAME:
+            snprintf(str, size, "%s", vars->media_file_name);
+            break;
+        case MARLIN_VAR_FILEPATH:
+            snprintf(str, size, "%s", vars->media_file_path);
             break;
         default:
-            sprintf(str, "???");
+            snprintf(str, size, "???");
```

Comment:
commit 472123a
in real this cannot happened because filename and filepath is read only, but I fixed it. We plan replace marlin_server input character queue with same message queue used for clients, so all this code will be deleted, but we cannot manage this in 4.1.0 beta.